WooCommerce/Classes/ViewModels/Order Details/OrderDetailsDataSource.swift

```diff
@@ -1517,11 +1517,27 @@ extension OrderDetailsDataSource {
     }
 
     private func isEligibleForBackendReceipt(completion: @escaping (Bool) -> Void) {
-        guard !isEligibleForPayment else {
+        guard isOrderStatusEligibleForReceipt else {
             return completion(false)
         }
-        ReceiptEligibilityUseCase().isEligibleForBackendReceipts { isEligibleForReceipt in
-            completion(isEligibleForReceipt)
+        let receiptEligibility = ReceiptEligibilityUseCase()
+
+        switch order.status {
+        case .completed, .processing, .refunded:
+            receiptEligibility.isEligibleForBackendReceipts { isEligibleForReceipt in
+                completion(isEligibleForReceipt)
+        }
```

Comment:
nit: the earlier indentation here might make it look like the switch statement ends here

```suggestion
            }
```